fuel/datasets/hdf5.py

```diff
@@ -328,35 +362,55 @@ def _file_handle(self):
 
     def get_data(self, state=None, request=None):
         if self.load_in_memory:
-            return self._in_memory_get_data(state=state, request=request)
+            rval, rval_shapes = self._in_memory_get_data(state, request)
         else:
-            return self._out_of_memory_get_data(state=state, request=request)
+            rval, rval_shapes = self._out_of_memory_get_data(state, request)
+        for i, shapes in enumerate(rval_shapes):
```

Comment:
Oh I see, you use it for an assignment further.

Perhaps, a bit of  refactoring/renaming would be good here, `i`, `j`, `rv` and `sh` is a bit too much of abbreviations and one-letter variables.